rholang/src/main/scala/coop/rchain/rholang/interpreter/ParBuilder.scala

```diff
@@ -13,30 +14,30 @@ import coop.rchain.rholang.syntax.rholang_mercury.{parser, Yylex}
 import coop.rchain.rholang.syntax.rholang_mercury.Absyn.Proc
 
 trait ParBuilder[F[_]] {
-  def buildNormalizedTerm(source: String): F[Par]
+  def buildNormalizedTerm(source: String, deployerPk: Option[PublicKey]): F[Par]
 
-  def buildNormalizedTerm(reader: Reader): F[Par]
+  def buildNormalizedTerm(reader: Reader, deployerPk: Option[PublicKey]): F[Par]
 
-  def buildPar(proc: Proc): F[Par]
+  def buildPar(proc: Proc, deployerPk: Option[PublicKey]): F[Par]
```

Comment:
Yeah, passing the deployerPk to the ParBuilder raises eyebrows... I guess next steps in future PRs are:
 - having a RhoEnv data structure passed with the deployerPk and deployId (deploy signature)
 - maybe making RhoEnv a trait, and abstracting away places where non-rholang things leak into rholang.

For now though, this is precisely the result I asked for and I find it much better than the previous version (which also had this issue).

Reply:
In the deploy id PR I am introducing `NormailizerEnv`.

But I don't get your second point about abstracting away places where non-rholang things leak into rholang. Can you elaborate?

Reply:
We'd like rholang to stay pristine and devoid of any usage-specific trivia, like the deployer id or deploy id. From rholang's perspective, if anything, one can imagine a use case for having compile-time provided values, and maybe runtime-provided values as well. Those are usually provided in Map-s dubbed Environments ;)

IOW, rholang should not know there is a 'deploy id' or a 'deployer id' or even their types. Rather, it should know that there will be some values injected at compile time, and various plugins to rholang (like the one providing support for the deployId name) can resolve those values using the string keys only the plugin need to know. I mean, deployId support is not a plugin now. But should be, if we're to make rholang pristine yet again. Point 2) is for long-term consideration though :)